include/libdnf/advisory/advisory_query.hpp

```diff
@@ -0,0 +1,114 @@
+
+    AdvisoryQuery & ifilter_type(const Advisory::Type type, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+    AdvisoryQuery & ifilter_type(
+        const std::vector<Advisory::Type> & types, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+
+    AdvisoryQuery & ifilter_reference(
+        const std::vector<std::string> & reference_id,
+        std::vector<AdvisoryReferenceType> types,
+        sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+    //TODO(amatej): What about other reference fitlers (title, url?) - do we want them?
+    AdvisoryQuery & ifilter_CVE(const std::string & pattern, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+    AdvisoryQuery & ifilter_CVE(
+        const std::vector<std::string> & patterns, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+    AdvisoryQuery & ifilter_bug(const std::string & pattern, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+    AdvisoryQuery & ifilter_bug(
+        const std::vector<std::string> & patterns, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+
+    AdvisoryQuery & ifilter_severity(const std::string & pattern, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+    AdvisoryQuery & ifilter_severity(
+        const std::vector<std::string> & patterns, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+
+    AdvisoryQuery & ifilter_applicability(const bool value = true);
+
+    /// Filter out advisories that don't contain at least one package fulfilling the cmp_type condition when compared to input package
+    ///
+    /// @param package      libdnf::rpm::package used when filtering.
+    /// @param cmp_type     Condition to fulfill with package.
+    /// @return This AdvisoryQuery with applied filter.
+    AdvisoryQuery & ifilter_package(
+        const libdnf::rpm::Package & package, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+
+    /// Filter out advisories that don't contain at least one package fulfilling the cmp_type condition when compated to input packages
+    ///
+    /// @param package_set  libdnf::rpm::PackageSet used when filtering.
+    /// @param cmp_type     Condition to fulfill with packages.
+    /// @return This AdvisoryQuery with applied filter.
+    AdvisoryQuery & ifilter_packages(
+        const libdnf::rpm::PackageSet & package_set, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+
+    std::vector<AdvisoryPackage> get_advisory_packages(
+        const libdnf::rpm::PackageSet & package_set, sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+
+    //TODO(amatej): create AdvisorySet (and its iterators), make AdvisoryQuery inherit AdvisorySet? (but its so many objects..)
+    std::vector<Advisory> get_advisories() const;
+    std::vector<AdvisoryPackage> get_sorted_advisory_packages(bool only_applicable = false) const;
+    std::vector<AdvisoryPackage> get_advisory_packages_from_applicable_collections();
```

Comment:
Methods `get_sorted_advisory_packages` and `get_advisory_packages_from_applicable_collections()` sounds like redundant. What about to remove one of them or make it as private.